corepack: add page (#6485)
tldr-pages/tldr: pages/common/corepack.md

# corepack

> Zero-runtime-dependency package acting as bridge between Node projects and their package managers.
> More information: <https://github.com/nodejs/corepack>.

> - Add the Corepack shims to the Node.js installation directory adding them as global commands:

Reviewer: What do you think about this?
```suggestion
- Add the Corepack shims to the Node.js installation directory to make them available as global commands:
```

`corepack enable`

- Add the Corepack shims to a specific directory:

`corepack enable --install-directory {{path/to/directory}}`

- Remove the Corepack shims from the Node.js installation directory:

`corepack disable`

> - Prepare a specific package manager:

Reviewer: What exactly does this do? 
Not sure if this is correct, but it sounds a bit better:
```suggestion
- Prepare Corepack for a specific package manager:
```

Author: `node@16` comes with `npm@7`, you can use `corepack prepare` for this:
```bash
$ corepack prepare npm@6.14.15
$ node -v
v16.9.0
$ npm -v
7.21.1
$ corepack npm -v
6.14.15
```

Reviewer 2: Not sure I understand @navarroaxel

Author: `corepack` currently supports `npm`, `pnpm` and `yarn`, you can "prepare" a specific version of this package managers to work with a given project. Maybe the project needs one of the following: `npm@6`, `npm@7`, `yarn@1` or `yarn@2`, with `corepack prepare` you set the correct binary to `corepack {{package_manager}}` to run `corepack npm ci`  or `corepack yarn outdated`, etc

Reviewer 2: What's the purpose of preparing a package manager for a given project though? Can't you use, like, `npm` directly for example?

Author: exactly you can install node (`node@16` comes with `npm@7` and you don't need to worry about `npm` or `yarn`:
```bash
$ sudo apt install nodejs
$ corepack yarn install
$ corepack prepare npm@6.14.15
$ corepack npm outdated # using npm v6
```

Reviewer 2: Oh I see, so it's a bit like `nvm` but for `npm`? I suppose but if there's some specific technical language = used in the source documentation we should use that.

Author: Exactly an `nvm` for packager managers.
```
corepack prepare [... name@version] 
This command will download the given package managers (or the one configured for the local project if no argument is passed in parameter) and store it within the Corepack cache. If the -o,--output flag is set (optionally with a path as parameter), an archive will also be generated that can be used by the corepack hydrate command.
```

> `corepack prepare {{package}}@{{version}} --active`

Reviewer: ```suggestion
`corepack prepare {{package_manager}}@{{version}} --active`
```
Hope this is correct.

- Prepare the package manager configured for the project in the current path:

`corepack prepare`

> - Use a package manager without add it as a global command:

Reviewer: This sentense doesn't seem very complete.

Author: how can you explain this?
```bash
$ corepack yarn install
```
Maybe the diff of make `yarn` as a global command of the 1st example is not needed here.

Reviewer 2: Perhaps this will read better?
```suggestion
- Use a package manager without installing it as a global command:
```
